### ISSUE TITLE
Show progress bar while trimming / muting video

### ISSUE THREAD

**Issue by spacedmonkey** [swissspidy/web-stories-wp#9158]
## Feature Description

While a uploading a video or while it is transcoding, a progress bar is shown. See screenshot.

<img width="248" alt="Screenshot 2021-09-27 at 14 37 41" src="https://user-images.githubusercontent.com/237508/134919697-6918eaf2-007c-4893-b6e1-95510fa67a71.png">

However, when a video is trimmed or muting, no progress bar is shown.

**Comment by spacedmonkey** [comment]
Related: #5982 #869

**Comment by swissspidy** [comment]
The code in question is here:

https://github.com/google/web-stories-wp/blob/25c3f33b97dddb6e93f379288a3fca30eafe53da/packages/story-editor/src/components/library/panes/media/common/mediaElement.js#L218-L220

Re-posting my question from Slack:

Obviously the progress bar is shown when videos are being optimized. So that means they're marked as `local`. Why isn't that the case for muting and trimming? That sounds like the bug that'd need to be fixed here.

**Comment by spacedmonkey** [comment]
As originally discussed this in [thread](https://xwp.slack.com/archives/C013J2EGH8R/p1632747289083500). 

Maybe the progress bar for transcoding / muting / trimming could be a different color ( maybe green? ) to indicate one progress bar has finished and another one has started.

**Comment by swissspidy** [comment]
I still think showing different colors or similar warrants its own ticket with appropriate UX feedback.

The issue is here is that no progress bar is shown for these items. However, we show a progress a bar when uploading media. Muting/Trimming causes new media to be uploaded, but there is no progress bar. So that's clearly a bug that could be easily fixed.

**Comment by gajasinski** [comment]
For clarification:

- why would you show different colors? If the vide is being optimized (whether muted, transcoding, or trimming), something is in progress. I'm not sure how much value there is in differentiating the colors.
- Not sure this should be green as green typically means the activity is complete....you could turn it green once it is finished, but keeping it green the entire time might be confusing.

@aaskedall cc

**Comment by spacedmonkey** [comment]
> why would you show different colors?

To indicate two different operation are running, one to processing and one for uploading. The current progress bar is blue, I don't mind reusing that, just want to make it clear. I will leave picking colours to UX people :P

**Comment by gajasinski** [comment]
Does it matter? Isn't uploading a completely separate action? Can you upload AND process at the same time? Do these happen concurrently?

**Comment by spacedmonkey** [comment]
>  Can you upload AND process at the same time? Do these happen concurrently?

No, processing is done first ( locally ) and then the uploading is done. You have to wait until the file is process before it can be uploaded. So they two process, run in sequence.

**Comment by swissspidy** [comment]
When uploading a new video, we process/optimize/transcode it first in the browser before actually uploading to the server. This is a super seamless process. I also don't think that we should artificially split this up into two processes by adding different colors. IMO all that matters to the user is that media is being processed. One progress bar with one color seems to sufficiently communicate this state.

**Comment by spacedmonkey** [comment]
Okay, let's not get hung up on the colour of the progress bar, just a idea that no one seems like to let's move past it. 

@barklund Also flagged the lack of progress bar https://github.com/google/web-stories-wp/pull/9142#pullrequestreview-764417172

If we are going to block saving / publishing story, it is useful for users to know what video is blocking this.

**Comment by swissspidy** [comment]
The lack of a progress bar is clearly a bug so please feel free to move ahead and fix it. See https://github.com/google/web-stories-wp/issues/9158#issuecomment-927893529
